### ISSUE TITLE
azurerm_linux_function_app requires both connection string and instrumentation key for app insights

### ISSUE THREAD

**Issue by yamada-grasys** [hashicorp/terraform-provider-azurerm#18455]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment

### Terraform Version

1.2.4

### AzureRM Provider Version

3.23

### Affected Resource(s)/Data Source(s)

azurerm_linux_function_app

### Terraform Configuration Files

```hcl
resource "azurerm_linux_function_app_slot" "example" {
  function_app_id             = "***"
  storage_account_access_key = "***"
  storage_account_name       = "***"
  name  = "staging"
  site_config {
    application_insights_connection_string  = "***"
#    application_insights_key = ""
  }

  timeouts {}
}
```

### Debug Output/Panic Output

```shell
| Error: Missing required argument
│ 
│   with azurerm_linux_function_app_slot.example
│   on main.tf line 9, in resource "azurerm_linux_function_app_slot" "example":
│    9:     application_insights_connection_string  = "***"
│ 
│ "site_config.0.application_insights_connection_string": all of
│ `site_config.0.application_insights_connection_string,site_config.0.application_insights_key`
│ must be specified
╵
```

### Steps to Reproduce

terraform plan

### References

azurerm_windows_function_app had same problem:
https://github.com/hashicorp/terraform-provider-azurerm/issues/16276

and this problem resolved:
https://github.com/hashicorp/terraform-provider-azurerm/pull/18343

but it seems that the problem on azurerm_linux_function_app don't resolved yet:
https://github.com/hashicorp/terraform-provider-azurerm/blob/d9283263cb7516dc97bbb4d6eb973ff5bc92da10/internal/services/appservice/helpers/function_app_slot_schema.go#L403
https://github.com/hashicorp/terraform-provider-azurerm/blob/d9283263cb7516dc97bbb4d6eb973ff5bc92da10/internal/services/appservice/helpers/function_app_slot_schema.go#L414

**Comment by xiaxyi** [comment]
Thanks @yamada-grasys for raising this issue, let me check the linux version. we'll remove the restrictions once confirmed the same behavior should be made to Linux

**Comment by xiaxyi** [comment]
@yamada-grasys you can track the status in the pr:
https://github.com/hashicorp/terraform-provider-azurerm/pull/18477

**Comment by CorrenSoft** [comment]
This issue has been resolved already, right?

**Comment by rcskosir** [comment]
@yamada-grasys Thanks for taking the time to submit this issue. It looks like this has been resolved as of #18477. As such, I am going to mark this issue as closed.
